axiom/nr_segment_message.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef NR_SEGMENT_MESSAGE_HDR
+#define NR_SEGMENT_MESSAGE_HDR
+
+#include "nr_segment.h"
+#include "nr_segment_traces.h"
+
+/*
+ * Note:
+ * CAT is EOLed and this feature is not compatible with CAT.
+ */
+
+typedef enum _nr_segment_message_destination_type_t {
+  NR_MESSAGE_DESTINATION_TYPE_QUEUE,
+  NR_MESSAGE_DESTINATION_TYPE_TOPIC,
+  NR_MESSAGE_DESTINATION_TYPE_TEMP_QUEUE,
+  NR_MESSAGE_DESTINATION_TYPE_TEMP_TOPIC,
+  NR_MESSAGE_DESTINATION_TYPE_EXCHANGE
+} nr_segment_message_destination_type_t;
+
+typedef struct {
+  /* All strings are null-terminated. When unset, the strings are ingored. */
```

Comment:
Does "unset" mean set to `NULL` or an empty string or both?

Reply:
unset means set to NULL, for message segment we further don't allow empty strings, so if we detect NULL or empty, we leave the attribute unset.

Reply:
So would it be accurate for the comment to say "When NULL, the strings are ignored"?  Or is it "NULL or empty"?

Reply:
Yes, NULL or empty.
please see: [925b94c](https://github.com/newrelic/newrelic-php-agent/pull/991/commits/925b94caf9fb5ce2999efcf57af0572f9a7938a0)